packages/style-loader/index.js

```diff
@@ -2,64 +2,103 @@ const ExtractTextPlugin = require('extract-text-webpack-plugin');
 const merge = require('deepmerge');
 
 module.exports = (neutrino, opts = {}) => {
+  const cssTest = neutrino.regexFromExtensions(['css']);
+  const cssModulesTest = neutrino.regexFromExtensions(['module.css']);
+
   const options = merge({
+    test: (input) => {
+      const isCssModule = cssModulesTest.test(input);
+      const isRegularCss = cssTest.test(input);
+
+      if (opts.modules !== false && isCssModule) {
+        return false;
+      }
+
+      return isRegularCss;
+    },
+    ruleId: 'style',
     styleUseId: 'style',
+    cssUseId: 'css',
     hot: true,
+    hotUseId: 'hot',
+    modules: true,
+    modulesSuffix: '-modules',
+    modulesTest: cssModulesTest,
+    extractId: 'extract',
     extract: {
       plugin: {
         filename: neutrino.options.command === 'build' ? '[name].[contenthash].css' : '[name].css'
       }
     }
   }, opts);
 
-  neutrino.config.module
-    .rule(options.ruleId || 'style')
-      .test(neutrino.regexFromExtensions(['css']))
-      .use(options.styleUseId)
-         .loader(require.resolve('style-loader'))
-         .when(options.style, use => use.options(options.style))
-         .end()
-      .use(options.cssUseId || 'css')
-        .loader(require.resolve('css-loader'))
-        .when(options.css, use => use.options(options.css));
-
-  if (options.extract) {
-    const styleRule = neutrino.config.module.rule('style');
-    const styleEntries = styleRule.uses.entries();
-    const useKeys = Object.keys(styleEntries).filter(key => key !== options.styleUseId);
+  const rules = [options];
 
-    options.extract.loader = Object.assign({
-      use: useKeys.map(key => ({
-        loader: styleEntries[key].get('loader'),
-        options: styleEntries[key].get('options')
-      })),
-      fallback: {
-        loader: styleEntries[options.styleUseId].get('loader'),
-        options: styleEntries[options.styleUseId].get('options')
+  if (options.modules) {
```

Comment:
Something to consider here might be to set `options.extract.plugin.ignoreOrder: options.modules`, just as a nice default: https://github.com/webpack-contrib/extract-text-webpack-plugin#options

Reply:
Ooo nice!

Reply:
Hrmm - `ignoreOrder` is an option for `options.extract.plugin`, not `options.css`.

Reply:
Yikes, sorry. :(

Reply:
I think we should be able to just do it up in the initial `options` merge, eg
```
    extract: {
      plugin: {
        filename: neutrino.options.command === 'build' ? '[name].[contenthash].css' : '[name].css',
        ignoreOrder: opts.modules !== false
      }
    }
```